{% extends 'base.html' %}
{% block content %}
{% load static %}
{% load site_tags %}
<div class="help_page">
<h1>GTalign-web server</h1>
<p>GTalign-web has been developed at the <a href="https://bioinformatics.lt">Department of Bioinformatics</a> of the <a href="http://www.ibt.lt">Institute of Biotechnology</a> (<a href="https://gmc.vu.lt">Life Sciences Center</a>, <a href="https://www.vu.lt">Vilnius University</a>).</p>
<p>This website is free and open to all users and there is no login requirement.</p>
<h2 id="browser_compatibility">Browser compatibility</h2>
<p>GTalign-web has been tested and functions correctly on the following browsers:</p>
<table class="table table-striped table-sm w-auto">
    <thead><tr>
        <td>OS</td>
        <td>Chrome</td>
        <td>Firefox</td>
        <td>Microsoft Edge</td>
        <td>Safari</td>
    </tr></thead>
    <tbody>
        <tr>
            <td>Linux</td>
            <td>135</td>
            <td>136</td>
            <td>n/a</td>
            <td>n/a</td>
        </tr>
        <tr>
            <td>MacOS</td>
            <td>--</td>
            <td>--</td>
            <td>n/a</td>
            <td>--</td>
        </tr>
        <tr>
            <td>Windows</td>
            <td>131</td>
            <td>133</td>
            <td>127</td>
            <td>n/a</td>
        </tr>
    </tbody>
</table>
<h2 id="data_and_software">Software and Data Availability</h2>
<p>
GTalign-web is fully open source.
The source code of the {{ 'chemikeris/comer_web'|github_link:"frontend web server" }} and the {{ 'minmarg/gtalign-web-backend'|github_link:"backend computation server" }} is freely available.
</p>
<p>
Standalone {{ "minmarg/gtalign_alpha"|github_link:"GTalign" }} software for GPU-accelerated rapid protein structure search and alignment is open source and is available for local use.
</p>
<p>
GTalign-web uses the protein structural data available from the following databases:
</p>
<ul>
	<li><a href="https://www.wwpdb.org/">PDB</a>: Protein Data Bank</li>
	<li><a href="https://scop.berkeley.edu/">SCOPe</a>: Structural Classification of Proteins (extended)</li>
	<li><a href="http://prodata.swmed.edu/ecod/">ECOD</a>: Evolutionary Classification of protein Domains</li>
	<li><a href="https://alphafold.ebi.ac.uk/">AlphaFold Database</a></li>
    <li><a href="https://bfvd.foldseek.com/">BFVD</a>: the Big Fantastic Virus Database</li>
</ul>
<p>Sequence-level protein data is collected from these databases:</p>
<ul>
	<li><a href="https://www.uniprot.org/">UniProtKB/Swiss-Prot</a>: Reviewed entries from the UniProt Knowledgebase</li>
	<li><a href="https://www.uniprot.org/proteomes">Reference proteomes</a>: A collection of 48 representative proteomes</li>
	<li><a href="https://uniclust.mmseqs.com/">UniRef30</a>: UniProt entries clustered at 30% sequence identity</li>
</ul>
<p>Benchmarking data for GTalign-web are available on {{ 'minmarg/gtalign-web-benchmark'|github_link:"GitHub" }}.</p>
<h2 id="cite_us">Cite us</h2>
<p>If GTalign is useful for your research, please cite our articles:<p>
<ul>
    <li>Dapkūnas J, Margelevičius M. Web-based GTalign: bridging speed and accuracy in protein structure analysis. <a href="https://doi.org/10.1093/nar/gkaf398">Nucleic Acids Res, gkaf398 (2025)</a>.</li>
	<li>Margelevičius M. GTalign: spatial index-driven protein structure alignment, superposition, and search. <a href="https://doi.org/10.1038/s41467-024-51669-z">Nature Communications 15, 7305 (2024)</a>.</li>
</ul>
<h2 id="privacy">Privacy policy</h2>
<p>The GTalign-web collects the user's email address, if the user wants to be informed when the job has finished.
The server also uses cookies to improve the user experience.
We do not track users, and some essential cookies are necessary for the proper functioning and security of the web server and cannot be disabled.
More detailed information can be found in the <a href="https://bioinformatics.lt/privacy/">privacy policy</a> page.</p>
<h2 id="contacts">Contacts</h2>
<p>For enquiries please email us at mindaugas.margelevicius@bti.vu.lt or justas.dapkunas@bti.vu.lt.
<h2 id="funding">Funding</h2>
<p>This work was supported by the Research Council of Lithuania (LMTLT).</p>
</div>
{% endblock %}
